src/pycita/pycita.py

```diff
@@ -0,0 +1,696 @@
+        """
+        根据区块hash获取区块详情.
+
+        :param hash: 32字节的hash
+        :param tx_detail: True 区块中会包含交易详情, 否则只包含交易hash
+        :return: 区块详情
+        """
+        hash_str = param_to_str(hash)
+        assert len(hash_str) == 64 + 2
+        r = self._jsonrpc('getBlockByHash', [hash_str, tx_detail])
+        return cast(Dict, r)
+
+    def get_block_by_number(self, height: int, tx_detail: bool = False) -> Dict:
+        """
+        根据区块id获取区块详情.
+
+        :param height: 区块高度, 从0起
+        :param tx_detail: True 区块中会包含交易详情, 否则只包含交易hash
+        :return: 区块详情
+        """
+        assert height >= 0
+        r = self._jsonrpc('getBlockByNumber', ['0x%02x' % height, tx_detail])
+        return cast(Dict, r)
+
+    def get_meta_data(self) -> Dict:
+        """
+        查询链上元数据.
+        """
+        r = self._jsonrpc('getMetaData', ['latest'])
+        return cast(Dict, r)
+
+    def send_raw_transaction(self, data: PARAM) -> str:
+        """
+        发送原始交易数据.
+
+        :param data: 待发送的数据.
+        :return: 交易hash.
+        """
+        r = self._jsonrpc('sendRawTransaction', [param_to_str(data)])
+        return cast(Dict, r)['hash']
+
+    def send_transaction(self, private_key: PARAM, to_addr: PARAM, code: PARAM, value: int = 0, quota: int = DEFAULT_QUOTA, max_wait_block: int = 88):
+        """
+        发送完整交易数据.
+
+        :param private_key: 私钥.
+        :param to_addr: 接收方地址. 如果是合约部署, 则为b''.
+        :param code: 字节码.
+        :param value: 金额.
+        :param quota: 调用配额.
+        :param max_wait_block: 交易至多等待多少个区块. 默认88.
+        :return: 签名后的bytes.
+        """
+        block_number = self.get_latest_block_number()
+        data = self.signer.make_raw_tx(param_to_bytes(private_key),
+                                       param_to_bytes(to_addr),
+                                       param_to_bytes(code),
+                                       block_number + max_wait_block, value, quota)
+        return self.send_raw_transaction(data)
+
+    def deploy_contract(self, private_key: PARAM, code: PARAM, param: PARAM = b'') -> str:
+        """
+        部署合约.
+
+        :param private_key: 私钥
+        :param code: 编译后的合约
+        :param params: 合约构造函数的参数经encode_param编码后的bytes. 无参数用 b''
+        :return: 交易回执地址
+        """
+        return self.send_transaction(private_key, b'', param_to_bytes(join_param(code, param)))
+
+    def confirm_transaction(self, hash_addr: PARAM, timeout: int = -1) -> Dict:
+        """
+        等待交易完成.
+
+        :param hash_addr: 回执地址, 32字节
+        :param timeout: 等待回执的时间, 单位秒. -1: 一直等待回执; 0: 无论是否达成共识, 直接返回; 其他值表示超时时间
+        :return: 回执结果.
+        """
+        r: Dict = {}
+        t0 = time.time()
+        r = self.get_transcation_receipt(hash_addr, 0)
+
+        # 先获得交易回执
+        while 'blockNumber' not in r:
+            time.sleep(1)
+            t1 = time.time()
+            if t1 - t0 >= timeout and timeout != -1:
```

Comment:
t1 - t0 >= timeout != -1, this expression will be elegant

Reply:
Improved.